runtime/src/main/java/io/quarkiverse/githubapp/GitHubCustomizer.java

```diff
@@ -21,4 +21,21 @@ public interface GitHubCustomizer {
      * @param builder to customize
      */
     void customize(GitHubBuilder builder);
+
+    /**
+     * Customize the {@link GitHubBuilder} with various options specifically for the application client. Default implementation
+     * delegates to {@link #customize(GitHubBuilder)} for backward compatibility.
+     * <p>
+     * Note that customizations should never use {@link GitHubBuilder#withAppInstallationToken(String)}, because this is the
+     * application client. Customizations that use {@link GitHubBuilder#withJwtToken(String)} and
+     * {@link GitHubBuilder#withEndpoint(String)} are eventually overridden by the service.
+     *
+     * To specify a custom endpoint, use configuration properties {@code quarkus.github-app.instance-endpoint} or
+     * {@code quarkus.github-app.rest-api-endpoint}.
+     *
+     * @param builder to customize
+     */
+    default void customizeApplicationClient(GitHubBuilder builder) {
+        customize(builder);
+    }
```

Comment:
Thanks a lot for the PR!

Maybe you can add some considerations about what you can't customize in `customize()` and add a pointer to `customizeApplicationClient` in the Javadoc of `customize()`?

Finally and a bit sorry about it, but I have an annoying eye condition and I would appreciate if you could remove the emojis from the description/commit message. I can't afford the commit history to look like a Christmas tree as it makes things quite painful for me :). Thanks!

Reply:
TBH I was even wondering if we should deprecate `customize()` and have two separate methods instead but it might end up being less practical for some use cases.

Reply:
I too was wondering about having two separate methods.

Instead of deprecating the `customize()` method, what if that method meant "customize both" and the separate methods default to calling `customize()`?

```java
void customize(GitHubBuilder builder);

default void customizeApplicationClient(GitHubBuilder builder) { customize(builder); }

default void customizeInstallationClient(GitHubBuilder builder) { customize(builder); }
```

It's a little awkward because the GitHubService would never actually call `customize`: it would only call `customizeApplicationClient` and `customizeInstallationClient`.

Reply:
Yeah, that's more or less what I had in mind.
But, let's start simple, we can still complicate things later if we have a proper use case.

So, in the end, I would stick with what you did with the only change being adding some Javadoc to `customize()` to explain in which case you should use the other method.

Does it work for you?

Reply:
Yes, just updated with that Javadoc change.

And I removed the emoji from commit message + PR summary.